### DISCUSSION TITLE
Is it possible to add a method to array prototype?

### DISCUSSION THREAD

**Discussion by heecheon92** [vercel/next.js#54042] [Help] [answered]
### Summary

Basically, I have been using some custom function on array by attaching function implementation like following.

array-extension.ts

```typescript
declare global {
  interface Array<T> {
    someFunction(): Array<T>;
  }
}

if (!Array.prototype.someFunction){
  Array.prototype.someFunction = function () {
     // doSomething
    return [...this]
  }
} 

export {}
```
then in "pages.tsx" (at the root)

import "./array-extension"

I've been using code above just fine in ReactNative and the method implementation never gets recognized during the runtime.

Lint doesn't complain that I cannot use as vscode also provides "someFunction" as autocomplete.

Whenever I try to use this "someFunction", the browser spits "someFunction is not a fuction"

Maybe I am missing something about web dev environment?

I understand that attaching an additional method to an existing prototype is not a desirable to do so but still possible.

How do I achieve above with NextJS with appDir=true?

**Comment by icyJoseph** [comment, marked as answer]
Just add `"use client"` on top of the file (`array-extension.ts`).

Also, it is probably a good idea to import it at the root layout, before anything else.

And it is also a good idea to define it via Object.defineProperty. For example: https://github.com/kevlatus/polyfill-array-includes/blob/master/array-includes.js

**Reply by heecheon92** [reply]
I will give it a shot. Thank you for the suggestion

**Reply by heecheon92** [reply]
I did add import the array-extension file at the top of the root layout but unfortunately couldn't access methods other than the root layout file.

**Reply by heecheon92** [reply]
adding "use client" at the top of the polyfill file worked. Thanks for the answer
